api/spec/src/productcatalog/subscription.tsp

```diff
@@ -0,0 +1,231 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/openapi";
+import "@typespec/openapi3";
+import "@typespec/versioning";
+
+import "../types.tsp";
+import "../errors.tsp";
+import "./prices.tsp";
+import "./ratecards.tsp";
+import "./discounts.tsp";
+import "../entitlements/main.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.OpenAPI;
+using TypeSpec.Versioning;
+
+namespace OpenMeter.ProductCatalog;
+
+/**
+ * Expanded subscription
+ */
+@friendlyName("SubscriptionExpanded")
+model SubscriptionExpanded {
+  ...Subscription;
+  phases: SubscriptionPhaseExpanded[];
+}
+
+/**
+ * Subscription is an exact subscription instance.
+ */
+@friendlyName("Subscription")
+model Subscription {
+  ...global.Resource;
+
+  /**
+   * The customer ID of the subscription.
+   */
+  customerId: ULID;
+
+  /**
+   * The base plan of the subscription.
+   */
+  plan: {
+    /**
+     * The plan Key of the subscription.
+     */
+    key: Key;
+
+    /**
+     * The plan version of the subscription.
+     */
+    version: integer;
+  };
+
+  /**
+   * The currency code of the subscription.
+   * Will be revised once we add multi currency support.
+   */
+  @summary("Currency")
+  currency: CurrencyCode = "USD";
+
+  /**
+   * If the subscription is canceled or otherwise have to end activeTo denotes the end date.
+   */
+  activeTo?: DateTime;
+}
+
+/**
+ * Expanded subscription phase
+ */
+@friendlyName("SubscriptionPhaseExpanded")
+model SubscriptionPhaseExpanded {
+  ...SubscriptionPhase;
+  items: SubscriptionItem[];
+}
+
+/**
+ * Subscription phase, analogous to plan phases.
+ */
+@friendlyName("SubscriptionPhase")
+model SubscriptionPhase {
+  ...VirtualResource;
+  ...OmitProperties<SubscriptionPhaseCreateInput, "startAfter">;
+
+  /**
+   * The time from which the phase is active.
+   */
+  activeFrom: DateTime;
+}
+
+@friendlyName("SubscriptionPhaseCreateInput")
+model SubscriptionPhaseCreateInput {
```

Comment:
```suggestion
model SubscriptionPhaseCreate {
```
So we are aligned with naming used by rest helpers like `TypeSpec.Rest.Resource.ResourceCreateModel` which normally should be used. (This one is an exception).